vnext/Microsoft.ReactNative.Cxx/JSI/JsiApi.h

```diff
@@ -7,8 +7,55 @@
 #include "../ReactContext.h"
 #include "JsiAbiApi.h"
 
+// Use __ImageBase to get current DLL handle.
+// http://blogs.msdn.com/oldnewthing/archive/2004/10/25/247180.aspx
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+
 namespace winrt::Microsoft::ReactNative {
 
+// Get JSI Runtime from the current JS dispatcher thread.
+// If it is not found, then create it and store it in the context.Properties().
+// Make sure that the JSI runtime holder is removed when the instance is unloaded.
+facebook::jsi::Runtime &GetOrCreateContextRuntime(ReactContext const &context) noexcept {
+  ReactDispatcher jsDispatcher = context.JSDispatcher();
```

Comment:
>ReactDispatcher [](start = 2, length = 15)

(nit) can this be a &?

Reply:
Good point! I will do that.

---
In reply to: [542903952](https://github.com/microsoft/react-native-windows/pull/6745#discussion_r542903952) [](ancestors = 542903952)

Reply:
The JSDispatcher() function returns a temporary: we cannot assign it to a reference.

---
In reply to: [543004717](https://github.com/microsoft/react-native-windows/pull/6745#discussion_r543004717) [](ancestors = 543004717,542903952)